bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java

```diff
@@ -133,6 +133,16 @@ public LedgerFragmentReplicator(BookKeeper bkc, ClientConfiguration conf) {
     private static final Logger LOG = LoggerFactory
             .getLogger(LedgerFragmentReplicator.class);
 
+    private boolean isEntryExpectedOnDeadBookie(final LedgerHandle lh, Long eId, Set<Integer> deadBookieIndex) {
```

Comment:
looks like this would work and might be the shortest path from where you are to where you want to get.

it's somewhat unfortunate you can't just configure the LedgerHandle to avoid the dead bookies when creating the write set.  there's almost support in there for doing that, but it isn't too flexible and ultimately optional at both the LedgerHandle and DistributionSchedule implementation levels.  see how bookie health info is almost useful.